scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/edf/EDFPolicyExtended.java

```diff
@@ -170,4 +152,67 @@ private static Date getEffectiveExpectedExecutionTime(InternalJob internalJob, D
         }
     }
 
+    private static class CompositeComparator implements Comparator<JobDescriptor> {
+
+        private List<Comparator<InternalJob>> comparators;
+
+        public CompositeComparator(Comparator<InternalJob>... comparators) {
+            this.comparators = asList(comparators);
+        }
+
+        @Override
+        public int compare(JobDescriptor job1, JobDescriptor job2) {
+            InternalJob internal1 = ((JobDescriptorImpl) job1).getInternal();
+            InternalJob internal2 = ((JobDescriptorImpl) job2).getInternal();
+
+            for (Comparator<InternalJob> comparator : comparators) {
+                final int result = comparator.compare(internal1, internal2);
+                if (result != 0) {
+                    return result;
+                }
+            }
+            return 0;
+        }
+    }
+
+    private static final Comparator<InternalJob> byPriority = Comparator.comparing(JobState::getPriority);
+
+    private static final Comparator<InternalJob> deadlineOverNoDeadline = (job1, job2) -> {
+        if (job1.getJobDeadline().isPresent() && !job2.getJobDeadline().isPresent()) {
+            return -1;
+        } else if (!job1.getJobDeadline().isPresent() && job2.getJobDeadline().isPresent()) {
+            return 1;
+        } else {
+            return 0;
+        }
+    };
+
+    private static final Comparator<InternalJob> startedOverNotStarted = (job1, job2) -> {
+        if (job1.getJobInfo().getStartTime() >= 0 && job2.getJobInfo().getStartTime() < 0) {
+            return -1;
+        } else if (job1.getJobInfo().getStartTime() < 0 && job2.getJobInfo().getStartTime() >= 0) {
+            return 1;
+        } else {
+            return 0;
+        }
+    };
+
+    private static final Comparator<InternalJob> byStartedTime = (job1, job2) -> {
+        if (job1.getJobInfo().getStartTime() >= 0 && job2.getJobInfo().getStartTime() >= 0) {
+            return Long.compare(job1.getJobInfo().getStartTime(), job2.getJobInfo().getStartTime());
+        } else {
+            return 0;
+        }
+    };
+
+    private static final Function<Date, Comparator<InternalJob>> byDistanceBetweenFinishAndDeadline = (
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "now" parameter [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1611)